R/conditions.R

```diff
@@ -130,6 +130,63 @@ captureStackTraces <- function(expr) {
 #' @include globals.R
 .globals$deepStack <- NULL
 
+# (Warning, this is pretty crazy)
+#
+# appendWithLimit lets us append an element to a list, but if the list starts to
+# exceed a limit, it will start eliding elements from the middle of the list.
+# At the point of elision, it will insert a placeholder integer that indicates
+# how many elements were elided.
+#
+# This is useful for storing deep stack traces, where we don't want to overwhelm
+# the user with a huge number of stack traces, but both the head and tail may
+# contain important information.
+#
+# Example:
+# lst <- list("a", "b", "c", "d")
+#
+# appendWithLimit(lst, "e", limit = 4)
+# # Result: list("a", 1L, "c", "d", "e")
+#
+# appendWithLimit(lst, "e", limit = 4, keep_head = 2L)
+# # Result: list("a", "b", 1L, "d", "e")
+appendWithLimit <- function(lst, x, limit, keep_head = 1L) {
+  # Sanity check that the parameters make sense
+  stopifnot(keep_head > 0)
```

Comment:
I'd consider a more inclusive name, possibly `elide_index`?

Reply:
`retain_first_n`?

Reply:
Oh, `elide_position` meaning, elide starting at that position. That's not bad

Reply:
I tried `elide_position` but it's awkward for callers. I think "how many elements at the front do you want to keep?" is more natural to think about than "from what position do you want to start performing elision?"

Reply:
`retain_first_n` seems like an improvement to me, or even `keep_first_n`. I mainly want to avoid `head`.